geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/IngestCommand.scala

```diff
@@ -238,47 +238,89 @@ object IngestCommand extends LazyLogging {
 
+    // if there is no converter passed in
     if (converter == null && inputs.nonEmpty) {
-      // if there is no converter passed in, try to infer the schema from the input files themselves
-      Command.user.info("No converter defined - will attempt to detect schema from input files")
-      val file = inputs.iterator.flatMap(PathUtils.interpretPath).headOption.getOrElse {
-        throw new ParameterException("Parameter <files> did not evaluate to anything that could be read")
+      var open: () => InputStream = null
+      var file: FileHandle = null
+
+      // try to infer the schema from stdin
+      if (inputs == Inputs.StdInInputs) {
+        Command.user.info("No converter defined - will attempt to detect schema from stdin")
+
+        // Cache the bytes from the input stream, because we'll need them again for the ingest
+        val inputBytes = System.in.readAllBytes()
+
+        // Function that returns an input stream containing the data from stdin
+        def openStdin(): InputStream = {
+          try {
+            // This prevents the schema and type inference code from consuming all the bytes from stdin and leaving
+            // no bytes for the actual ingest to read
+            System.setIn(new ByteArrayInputStream(inputBytes))
+
+            // Return stdin
+            System.in
+          } catch {
+            case _: Exception => throw new ParameterException("Unable to read from stdin")
+          }
+        }
+
+        open = openStdin
       }
-      val opened = ListBuffer.empty[CloseableIterator[InputStream]]
-      def open(): InputStream = {
-        val streams = file.open.map(_._2)
-        opened += streams
-        if (streams.hasNext) { streams.next } else {
+      // try to infer the schema from the input files themselves
+      else {
+        Command.user.info("No converter defined - will attempt to detect schema from input files")
+        file = inputs.iterator.flatMap(PathUtils.interpretPath).headOption.getOrElse {
           throw new ParameterException("Parameter <files> did not evaluate to anything that could be read")
         }
+        val opened = ListBuffer.empty[CloseableIterator[InputStream]]
+
+        def openFile(): InputStream = {
+          val streams = file.open.map(_._2)
+          opened += streams
+          if (streams.hasNext) {
+            streams.next
+          } else {
+            throw new ParameterException("Parameter <files> did not evaluate to anything that could be read")
+          }
+        }
+
+        open = openFile
       }
+
       val (inferredSft, inferredConverter) = try {
         val opt = format match {
-          case None      => SimpleFeatureConverter.infer(open, Option(sft), Option(file.path))
-          case Some(fmt) => TypeAwareInference.infer(fmt, open, Option(sft), Option(file.path))
+          case None => SimpleFeatureConverter.infer(open, Option(sft))
+          case Some(fmt) => TypeAwareInference.infer(fmt, open, Option(sft), None)
```

Comment:
I think we need to keep the file.path here, probably filtered appropriately for stdin. some converters rely on the file path as well as the input stream.